libs/langchain-azure-cosmosdb/src/chat_histories/mongodb.ts

```diff
@@ -150,6 +165,64 @@ export class AzureCosmosDBMongoChatMessageHistory extends BaseListChatMessageHis
   async clear(): Promise<void> {
     await this.initialize();
 
-    await this.collection.deleteOne({ [ID_KEY]: this.sessionId });
+    await this.collection.deleteOne({
+      [ID_KEY]: this.sessionId,
+      [ID_USER]: this.userId,
+    });
+  }
+
+  async getAllSessions(): Promise<ChatSessionMongo[]> {
+    await this.initialize();
+    const documents = await this.collection.find({
+      [ID_USER]: this.userId,
+    }).toArray();
+
+    const chatSessions: ChatSessionMongo[] = documents.map((doc) => ({
+      id: doc[ID_KEY],
+      user_id: doc[ID_USER],
+      context: doc.context || {},
+    }));
+
+    return chatSessions;
+  }
+
+  async clearAllSessions() {
+    await this.initialize();
+    try {
+      await this.collection.deleteMany({
+        [ID_USER]: this.userId,
+      });
+    } catch (error) {
+      console.error("Error clearing chat history sessions:", error);
+      throw error;
+    }
+  }
+
+  async getContext(): Promise<Record<string, unknown>> {
+    await this.initialize();
+
+    const document = await this.collection.findOne({
+      [ID_KEY]: this.sessionId,
+      [ID_USER]: this.userId,
+    });
+    this.context = document?.context || this.context;
+    return this.context;
+  }
+
+  async setContext(context: Record<string, unknown>): Promise<void> {
+    await this.initialize();
+
+    try {
+      await this.collection.updateOne(
+        { [ID_KEY]: this.sessionId },
```

Comment:
Do we need `[ID_USER]: this.userId,` here too?

I assume this is sufficient in this case to just pass id?